main.tf

```diff
@@ -44,7 +44,7 @@ data "aws_iam_policy_document" "default" {
 
 resource "aws_iam_policy" "default" {
   count       = module.this.enabled && var.policy_document_count > 0 ? 1 : 0
-  name        = module.this.id
+  name        = var.policy_name ? var.policy_name : module.this.id
```

Comment:
@karinatitov you'll need to add a `variable` block to utilize a new variable this. e.g.: 

```hcl
variable "policy_name" {
  type             = string
  description = "... "
  default        = null 
} 
```